ls1intum/Artemis
# Add course reload button and automatic quiz display (visible)

### Checklist

### Motivation and Context
This is a performance improvement. Some students tend reload the whole Artemis application if they do not see certain exercises or lectures. However, this can lead to performance problems, especially when > 1000 students reload the browser in lectures at the same time.

### Description
This PR adds a for dashboard REST call for a single course and adds a reload button that triggers this call. This should reduce the performance impact on the server significantly compared to a browser reload. In addition, quizzes should be displayed automatically (using web sockets), when they are set to visible

- [x] A quiz which is visible, but not yet started (without start time), should be displayed at the top of the exercise list

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Overview. Double check that everything is shown correctly (and nothing is missing)
3. Navigate into one course
4. Click the reload button (student) --> nothing should change
5. Add a new exercises or lecture in a separate browser window / tab (instructor)
6. Click on the reload button: the exercise or lecture should be shown now (student)
7. Set a quiz without start time to visible (instructor)
8. The quiz should appear and be shown at the top (student)
